s2n.mk

```diff
@@ -169,6 +169,12 @@ ifeq ($(TRY_COMPILE_CPUID), 0)
 	DEFAULT_CFLAGS += -DS2N_CPUID_AVAILABLE
 endif
 
+# Determine if __attribute__((fallthrough)) is available
+TRY_COMPILE_FALL_THROUGH := $(shell echo "\#define FALL_THROUGH __attribute__((fallthrough))\nint main()\n {\nchar val = 'A';\nswitch (val) {\ncase 'A' :\nval++;\nFALL_THROUGH;\ncase\n'B' :\nbreak;\ndefault :\nbreak;\nreturn 0;\n}" | $(CC) -Werror -o test_fall_through.o -xc - > /dev/null 2>&1; echo $$?; rm test_fall_through.o > /dev/null 2>&1)
```

Comment:
I wonder if we should create a new directory with these probe snippets so they can more easily shared between the build systems. Then you could just do `$(shell cat ./features/fallthrough.c | $(CC))`. This is really hard to read otherwise :)